src/components/stationsSelect.tsx

```diff
@@ -36,6 +38,87 @@ export function StationSelect(props: Props) {
   } = props;
 
   const [open, setOpen] = useState(false);
+  const [dropDownCurrentState, setDropDownCurrentState] = useState(<></>)
+  const [allStationsLoaded, setAllStationsLoaded] = useState(false)
+  
+  //loading all stations into the dropdown (takes longer so only done if the user clicks on the "load all stations" button)
+  function loadAllStations() {
+    if(!availableStations) return
+    setAllStationsLoaded(true)
+    setDropDownCurrentState(
+      <CommandGroup>
+        {availableStations.map((station) => (
+          <CommandItem
+            key={station.id}
+            value={station.name}
+            onSelect={(currentValue) => {
+              const selectedStation = availableStations.find(
+                (station) =>
+                  station.name.toLowerCase() === currentValue,
+              );
+
+              setOpen(false);
+              setAllStationsLoaded(false)
+              if (selectedStation) onSelect(selectedStation);
+            }}
+          >
+            <Check
+              className={cn(
+                "mr-2 h-4 w-4",
+                selectedStations.find(
+                  (selected) => selected.id == station.id,
+                )
+                  ? "opacity-100"
+                  : "opacity-0",
+              )}
+            />
+            {station.name}
+          </CommandItem>
+        ))}
+      </CommandGroup>
+    )
+  }
+
+  //runs everytime the dropdown is opened
+  React.useEffect(() => {
+    if(!availableStations) return
+    setAllStationsLoaded(false)
+    //setting the dropdown to the first 800 stations to achieve a faster opening of the dropdown
+    setDropDownCurrentState(
```

Comment:
I dont think storing that much html in the state is a good idea. I would just populate `availableStations` with only 800 entries from the start (or introduce a new variable and call that `availableStationsTrimmed`. then once the button is clicked, we change the list to be all stations

Reply:
Is the updated version OK for you?